chromadb/segment/impl/manager/local.py

```diff
@@ -140,8 +154,41 @@ def delete_segments(self, collection_id: UUID) -> Sequence[UUID]:
         "LocalSegmentManager.get_segment",
         OpenTelemetryGranularity.OPERATION_AND_SEGMENT,
     )
+    def _get_segment_disk_size(self, collection_id: UUID) -> int:
+        segments = self._sysdb.get_segments(collection=collection_id, scope=SegmentScope.VECTOR)
+        if len(segments) == 0:
+            return 0
+        size = get_directory_size(
+            os.path.join(self._system.settings.require("persist_directory"), str(segments[0]["id"])))
+        return size
+
+    def _cleanup_segment(self, collection_id: UUID, target_size: int):
+        segment_sizes = {id: self._get_segment_disk_size(id) for id in self._segment_cache if
+                         SegmentScope.VECTOR in self._segment_cache[id]}
+        total_size = sum(segment_sizes.values())
+        new_segment_size = self._get_segment_disk_size(collection_id)
+
+        while total_size + new_segment_size >= target_size and self._segment_cache.keys():
```

Comment:
Can we move this LRU logic into a cleaner abstraction with eviction handles - like https://github.com/chroma-core/chroma/blob/main/chromadb/utils/lru_cache.py